daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -299,11 +299,42 @@ object SExpr {
     }
   }
 
+  /** A (single) let-expression with an unhungry, saturated builtin-application as RHS */
+  final case class SELet1BuiltinArithmetic(
+      builtin: SBuiltinArithmetic,
+      args: Array[SExprAtomic],
+      body: SExpr,
+  ) extends SExpr
+      with SomeArrayEquals {
+    def execute(machine: Machine): Unit = {
+      val arity = builtin.arity
+      val actuals = new util.ArrayList[SValue](arity)
+      var i = 0
+      while (i < arity) {
+        val arg = args(i)
+        val v = arg.lookupValue(machine)
+        actuals.add(v)
+        i += 1
+      }
+      builtin.executePure(actuals) match {
+        case Right(value) =>
+          machine.pushEnv(value) //use pushEnv not env.add so instrumentation is updated
+          machine.ctrl = body
+        case Left(message) =>
+          machine.traceLog.add("ArithmeticError:" + message, machine.lastLocation)
+          unwindToHandler(machine, SBuiltinException.Arithmetic)
```

Comment:
Did we decide it's ok to unconditionally throw a catchable exception?
i.e  LF which predates `minException` will now throw catchable exceptions.

Reply:
For what I remember, we said it was OK as long we do not create node with `version < minException` under a RollBack node

Reply:
Should be fine. It is slightly unfortunate that this means that just looking a function doesn’t tell me whether I can catch the exceptions it throws or not but given that we cannot avoid the dynamic nature in general, I think we don’t gain anything by handling this part differently.